README.md

```diff
@@ -216,6 +216,35 @@ The open and close callbacks come before and after the entire retry in the simpl
 
 Note that when there is more than one listener, they are in a list, so there is an order. In this case open will be called in the same order while onError and close will be called in reverse order.
 
+### Listeners for reflective method invocations
+
+When dealing with methods annotated with `@Retryable` or with Spring AOP intercepted methods,
+spring-retry provides the possibility to inspect in detail the method invocation within the
+`RetryListener` implementation. Such a scenario could be particularly useful when there is a need
+to monitor how often a certain method call has been retried and expose it with detailed tagging
+information (e.g. : class name, method name, or even parameter values in some exotic cases).
+
+All that needs to be done is checking whe
+
+```java
+
+template.registerListener(new MethodInvocationRetryListenerSupport() {
+      @Override
+      protected <T, E extends Throwable> void doClose(RetryContext context,
```

Comment:
Can't we just store a `MethodInvocation` in the `RetryContext` like we do with the `label` for example?

Reply:
@artembilan if you follow my comments (as well as what @dsyer wrote) from #119 using the mapping mechanism from the `RetryContext` was initially my thought as well.
`RetryContext` is stored in the `ThreadLocal` so I wouldn't see any big issues from this approach.

Nevertheless, `RetryContext` has the downside that it does not offer typed getters for the properties it holds (see `org.springframework.core.AttributeAccessor#getAttribute`).

IMO using a strongly typed concrete implemenation of `org.springframework.retry.RetryCallback` has the main advantage that the `RetryListener` will be somehow safer to implement.

Also worth noting is the fact that the `MethodInvocation` makes sense only for reflective method invocations (see `RetryOperationsInterceptor` and `StatefulRetryOperationsInterceptor`) - so adding the constant name to the `RetryContext` would be not very appropriate.